cpp/src/join/joining.cu

```diff
@@ -533,6 +534,70 @@ gdf_error join_call_compute_df(
     return GDF_SUCCESS;
   }
 
+
+  //if the inputs are nvcategory we need to make the dictionaries comparable
+  gdf_column ** new_left_cols = new gdf_column * [num_left_cols];
+  gdf_column ** new_right_cols = new gdf_column * [num_right_cols];
+  for(int column_index = 0; column_index < num_left_cols; column_index++){
+	  new_left_cols[column_index] = left_cols[column_index];
+  }
+  for(int column_index = 0; column_index < num_right_cols; column_index++){
+	  new_right_cols[column_index] = right_cols[column_index];
+  }
+  std::vector<gdf_column *> nv_category_copied_columns;
+  for(int join_column_index = 0; join_column_index < num_cols_to_join; join_column_index++){
+	  gdf_column * left_original_column = new_left_cols[left_join_cols[join_column_index]];
+	  gdf_column * right_original_column = new_right_cols[right_join_cols[join_column_index]];
+
+	  if(left_original_column->dtype == GDF_STRING_CATEGORY){
+		  if(right_original_column->dtype == GDF_STRING_CATEGORY){
+        gdf_column * new_join_columns[2];
+        new_join_columns[0] = left_original_column;
+        new_join_columns[1] = right_original_column;
+
+			  gdf_column * new_left_column = new gdf_column;
+			  gdf_column * new_right_column = new gdf_column;
+			  nv_category_copied_columns.push_back(new_left_column);
+			  nv_category_copied_columns.push_back(new_right_column);
+
+			  gdf_column * input_join_columns_merge[2];
+			  input_join_columns_merge[0] = left_original_column;
+			  input_join_columns_merge[1] = right_original_column;
+
+			  gdf_column_view(new_left_column, nullptr, nullptr, left_original_column->size, GDF_STRING_CATEGORY);
+			  gdf_column_view(new_right_column, nullptr, nullptr, right_original_column->size, GDF_STRING_CATEGORY);
+
+			  int col_width;
+			  get_column_byte_width(new_left_column, &col_width);
+			  RMM_TRY( RMM_ALLOC((void**)&(new_left_column->data), col_width * left_original_column->size, 0) ); // TODO: non-default stream?
```

Comment:
The `void**` cast is no longer necessary.

Reply:
@jrhemstad just to be sure, the idea is that the allocation for any pointer to data does not need the conversion, however other allocations of different type of data would need it, right? (for example the valids)

Reply:
All I was saying is that you can do this:

```
RMM_ALLOC(&(new_left_column->data),
```

You don't need the `(void**)`